apps/jobs/views.py

```diff
@@ -389,117 +396,88 @@ def leaderboard(request, challenge_phase_split_id):
+            },
+            {
+                "id": 2,
+                "name": "Molly Phase",
+                "start_date": "2018-10-28T14:22:53Z",
+                "end_date": "2020-06-19T14:22:53Z",
+                "message": {
+                    "message": "You have exhausted this month's submission limit!",
+                    "remaining_time": "1481076.929224"
+                }
             }
-        return Response(response_data, status=status.HTTP_200_OK)
-
-    # Checks if #today's successful submission is greater than or equal to max submission per day
-    elif submissions_done_today_count >= max_submissions_per_day_count:
-        date_time_now = timezone.now()
-        date_time_tomorrow = date_time_now + datetime.timedelta(1)
-        # Get the midnight time of the day i.e. 12:00 AM of next day.
-        midnight = date_time_tomorrow.replace(hour=0, minute=0, second=0)
-        remaining_time = midnight - date_time_now
-
-        response_data = {
-            'message': 'You have exhausted today\'s submission limit!',
-            'remaining_time': remaining_time
-        }
-        return Response(response_data, status=status.HTTP_200_OK)
-
+        ]
+    }
+    '''
+    phases_data = {}
+    challenge = get_challenge_model(challenge_pk)
+    if is_user_a_host_of_challenge(request.user, challenge_pk):
+        challenge_phase = ChallengePhase.objects.filter(
+            challenge=challenge).order_by('pk')
     else:
-        # calculate the remaining submissions from total submissions.
-        remaining_submission_count = max_submissions_count - submissions_done_count
-        # Calculate the remaining submissions for current month.
-        remaining_submissions_this_month_count = (max_submissions_per_month_count -
-                                                  submissions_done_this_month_count)
-        # Calculate the remaining submissions for today.
-        remaining_submissions_today_count = (max_submissions_per_day_count -
-                                             submissions_done_today_count)
-
-        remaining_submissions_this_month_count = min(remaining_submission_count,
-                                                     remaining_submissions_this_month_count)
-        remaining_submissions_today_count = min(remaining_submissions_this_month_count,
-                                                remaining_submissions_today_count)
-
-        response_data = {
-            'remaining_submissions_this_month_count': remaining_submissions_this_month_count,
-            'remaining_submissions_today_count': remaining_submissions_today_count,
-            'remaining_submissions': remaining_submission_count
-        }
-        return Response(response_data, status=status.HTTP_200_OK)
+        challenge_phase = ChallengePhase.objects.filter(
+            challenge=challenge, is_public=True).order_by('pk')
+
+    phases_data['participant_team'] = get_participant_team_name_of_user_for_a_challenge(
```

Comment:
Can we please make a single util function that gives us the participant team object? Both `get_participant_team_name_of_user_for_a_challenge()` and `get_participant_team_id_of_user_for_a_challenge()` can be removed then.